core/dbt/events/functions.py

```diff
@@ -16,9 +16,14 @@
 import numbers
 import os
 import uuid
-from typing import Any, Callable, Dict, List, Optional, Set, Tuple, Union
+from typing import Any, Callable, Dict, List, Optional, Set, Tuple, Union, Deque
 from dataclasses import _FIELD_BASE  # type: ignore[attr-defined]
+from collections import deque
 
+# create the global event history buffer with a max size of 100k records
+# TODO: make the maxlen something configurable from the command line via args(?)
+global EVENT_HISTORY
+EVENT_HISTORY: Deque = deque(maxlen=100000)
```

Comment:
@kwigley is this going to fit the bill for what's needed in Client/Server?  It should make a globally available deque of events so you can do whatever's needed.

Reply:
awesome! Ya, I believe so.